### PR TITLE
Replace some uses of size_t

### PR DESCRIPTION
**Pull request by MarcelHB**
## Description
See #1616, there were two errors and I noticed a bunch of stuff around it that could be type-wired and cleaned with `uint32_t` at the end.

## Checklist

- [x] Commit messages are descriptive and explain the rationale for changes
- [x] I used the same coding style as the surrounding code 
- [x] I have tested the proposed changes
- [x] I extended the documentation, if necessary
- [ ] The proposed change builds also on our build bots (check after submission)

### CODE REVIEW

File: gemrb/plugins/2DAImporter/2DAImporter.cpp [hunk 1]
```diff
@@ -102,7 +102,9 @@ bool p2DAImporter::Open(DataStream* str)
 			row++;
 		}
 	}
+
 	delete str;
+	assert(rows.size() < std::numeric_limits<index_t>::max());
```
Comment (lynxlynxlynx) [inline gemrb/plugins/2DAImporter/2DAImporter.cpp:107]: I don't think we need to be this paranoid. You'd need specially crafted data and tons of free memory to be able to fail here.

Comment (MarcelHB) [reply gemrb/plugins/2DAImporter/2DAImporter.cpp:107]: Force of habit when it comes to _all input is evil_ :shrug:

**Review by bradallred — APPROVED** [review]
this all looks good to me

**Comment by MarcelHB** [comment]
No merge button for me 🤷

**Pull request merged**